### PR TITLE
Jonmv/long deployment pipelines 2

### PR DESCRIPTION
**Pull request by jonmv**
@mpolden please review. This is just some additional scenario tests, with a fix to an unlikely deadlock scenario, by ensuring production tests always agree with their previous deployments on what versions to run.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentStatus.java [hunk 1]
```diff
@@ -345,31 +346,70 @@ private static Deployment withChange(Deployment deployment, Change change) {
...
+        }
 
         Optional<Instant> platformReadyAt = step.dependenciesCompletedAt(change.withoutApplication(), Optional.of(job));
         Optional<Instant> revisionReadyAt = step.dependenciesCompletedAt(change.withoutPlatform(), Optional.of(job));
 
-        // If the revision is not ready for this job, we prioritise the upgrade, assuming it was triggered first,
-        // unless this is a production test job, and the upgrade is already failing between deployment and here,
-        // in which case we must abandon the production test on the pure upgrade, so the revision can be deployed.
+        // If neither change is ready, we guess based on the specified rollout.
+        if (platformReadyAt.isEmpty() && revisionReadyAt.isEmpty()) switch (rollout) {
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentStatus.java:395]: Surprising choice of formatting. Consider adding newline/braces for multi-line if-body. 🙂

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTriggerTest.java [hunk 2]
```diff
@@ -1311,6 +1311,116 @@ public void testRevisionJoinsUpgradeWithSeparateRollout() {
...
+                     tester.jobs().last(app.instanceId(), productionUsEast3).get().versions());
+        app.runJob(productionUsEast3).triggerJobs()
+                .jobAborted(productionUsWest1).runJob(productionUsWest1);
+        tester.clock().advance(Duration.ofHours(1));
+        app.runJob(testUsEast3);
+        assertEquals(Change.empty(), app.instance().change());
+    }
+
+    @Test
+    public void testProductionTestNotBlockingDeploymentWithSimultaneousRollout() {
```
Comment (mpolden) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTriggerTest.java:1363]: Good catch!

**Review by mpolden — APPROVED** [review]

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentStatus.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentStatus.java:395]: Oops.

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTriggerTest.java (hunk 2 above)
Comment (jonmv) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTriggerTest.java:1363]: Hehe.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentStatus.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentStatus.java:395]: Will do in next PPR :)

**Pull request merged**